openldap-mixin/targets.libsonnet

```
@@ -0,0 +1,102 @@
+local g = import './g.libsonnet';
+local prometheusQuery = g.query.prometheus;
+
+{
+  new(this): {
+    local variables = this.grafana.variables,
+    local panel = g.panel,
+
+    uptimeQuery:: 'openldap_monitored_object{job=~"$job", instance=~"$instance", dn="cn=Uptime,cn=Time,cn=Monitor"}',
+    referralsQuery:: 'openldap_monitor_counter_object{job=~"$job", instance=~"$instance", dn="cn=Referrals,cn=Statistics,cn=Monitor"}',
+    directoryEntriesQuery(interval):: 'increase(openldap_monitor_counter_object{job=~"$job", instance=~"$instance", dn="cn=Entries,cn=Statistics,cn=Monitor"}[$%s:])' % interval,
+    connectionsQuery(interval):: 'increase(openldap_monitor_counter_object{job=~"$job", instance=~"$instance", dn="cn=Current,cn=Connections,cn=Monitor"}[$%s:])' % interval,
+    waitersQuery(type):: 'openldap_monitor_counter_object{job=~"$job", instance=~"$instance", dn="cn=%s,cn=Waiters,cn=Monitor"}' % type,
+    networkConnectivityQuery(interval):: 'increase(openldap_dial{job=~"$job", instance=~"$instance"}[$%s:])' % interval,
+    pduProcessedQuery(interval):: 'increase(openldap_monitor_counter_object{job=~"$job", instance=~"$instance", dn="cn=PDU,cn=Statistics,cn=Monitor"}[$%s:])' % interval,
+    authenticationAttemptsQuery(interval):: 'increase(openldap_bind{job=~"$job", instance=~"$instance"}[$%s:])' % interval,
+    coreOperationsQuery(operation, interval):: 'increase(openldap_monitor_operation{job=~"$job", instance=~"$instance", dn="cn=%s,cn=Operations,cn=Monitor"}[$%s:])' % [operation, interval],
+    auxiliaryOperationsQuery(operation, interval):: 'increase(openldap_monitor_operation{job=~"$job", instance=~"$instance", dn="cn=%s,cn=Operations,cn=Monitor"}[$%s:])' % [operation, interval],
+    primaryThreadActivityQuery(type):: 'openldap_monitored_object{job=~"$job", instance=~"$instance", dn="cn=%s,cn=Threads,cn=Monitor"}' % type,
+    threadQueueManagementQuery(type):: 'openldap_monitored_object{job=~"$job", instance=~"$instance", dn="cn=%s,cn=Threads,cn=Monitor"}' % type,
+
+    uptime:
+      prometheusQuery.new(
+        '${' + variables.datasources.prometheus.name + '}',
+        self.uptimeQuery
+      )
+      + prometheusQuery.withLegendFormat('{{instance}}'),
```

Most likely you should probably replace the bottom with . I would do this for all your targets with instance in them. This was a suggestion given previously by vitaly on my last mixin. You will also need to import it at the top like this `local commonlib = import 'common-lib/common/main.libsonnet';
`
```suggestion
            + prometheusQuery.withLegendFormat('%s' % commonlib.utils.labelsToPanelLegend(this.config.instanceLabels)),

```

---

updated